packages/funding-rate-proposer/index.js

```diff
@@ -8,6 +8,7 @@ const {
   waitForLogger,
   delay,
   FinancialContractFactoryClient,
+  OptimisticOracleClient,
```

Comment:
so now the `funding-rate-proposer` and the `optimisic-oracle` both import the `OptimisticOracleClient` and use it in somewhat diffrent ways. I was originally going to say that I think that this should stay within the domain of the `optimistic-oracle` but after more thought it does make sence to keep it here within the `funding-rate-proposer` as this bot's job is to monitor the funding rates of perps, not just OO enquired prices. 

Despite this small mixing of concerns, I think it makes sneak as a pattern.

Not really anything actionable here, just something I thought I'd raise. Thoughts?

Reply:
I think this highlights the problems with the `update() --> yourMethod()` pattern, where `update()` fetches unneccessary data for ALL OO requests/proposals/disputes whereas this bot only needs data coming from this specific bot.

However, it is nice that this bot doesn't have to determine which proposals have expired, instead it can make a simple `getSettleableProposals()` call to the OO client.

Reply:
ye, exactly. this is a pretty systemic issue with how we've built the client/implementor pattern. It would require a pretty drastic refactor to sort out and for now I think its' fine. Queries to Infura are cheap!